iothub_client/tests/common_e2e/iothubclient_common_e2e.c

```diff
@@ -287,12 +291,33 @@ static IOTHUBMESSAGE_DISPOSITION_RESULT ReceiveMessageCallback(IOTHUB_MESSAGE_HA
     return IOTHUBMESSAGE_ACCEPTED;
 }
 
+static void GenerateUniqueId(const char* prefix, size_t prefix_len, char* buffer, size_t buffer_len)
+{
+    if (prefix_len <= buffer_len)
+    {
+        memcpy(buffer, prefix, prefix_len);
+        buffer_len -= prefix_len;
+        if ((MSG_UNIQUE_ID_STAMP_LEN + 1) <= buffer_len)
+        {
+            buffer[prefix_len - 1] = '.';
```

Comment:
?
```suggestion
            buffer[prefix_len] = '.';
```

Reply:
prefix_len is the length of a string literal aka sizeof(""), so the -1 points to the null char and overrides it to a period

Reply:
nit would be to have a comment stating that is what's happening here since char* are usually given with length not including the `NULL`.